packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -91,7 +91,35 @@ export function BlockEditor( {
 		return () => window.removeEventListener( 'scroll', wpPinMenuEvent );
 	}, [] );
 
-	const settings = useMemo< Partial< ProductEditorSettings > >( () => {
+	const [ settingsGlobal, setSettingsGlobal ] = useState<
+		Partial< ProductEditorSettings > | undefined
+	>( undefined );
+
+	useEffect( () => {
+		let timeoutId: number;
+
+		const checkSettingsGlobal = () => {
+			if ( window.productBlockEditorSettings !== undefined ) {
+				setSettingsGlobal( window.productBlockEditorSettings );
+			} else {
+				timeoutId = setTimeout( checkSettingsGlobal, 100 );
+			}
+		};
+
+		checkSettingsGlobal();
+
+		return () => {
+			clearTimeout( timeoutId );
+		};
+	}, [] );
```

Comment:
This way to ensure loading the global settings looks interesting.  Do you know if there is an endpoint to request the setting?

Reply:
There is no endpoint currently. Certainly an endpoint would be a good future step. Some of the settings we put in the endpoint have been considered for separate endpoints (such as the product templates).

This is a minimal change to address the immediate issue.